### PR TITLE
fix 9939: docker fail to remove btrfs subvolumes when destroying container

### LINKED ISSUE #9939: docker does not remove btrfs subvolumes when destroying container
I receive the following error when deleting a container which created a btrfs subvolume (as happens when you run docker in docker).

```
# docker run --rm fedora:20 sh -c 'yum -y -q install btrfs-progs && btrfs subvolume create /test'
Public key for lzo-2.08-1.fc20.x86_64.rpm is not installed
Public key for e2fsprogs-libs-1.42.8-3.fc20.x86_64.rpm is not installed
Importing GPG key 0x246110C1:
 Userid     : "Fedora (20) <fedora@fedoraproject.org>"
 Fingerprint: c7c9 a9c8 9153 f201 83ce 7cba 2eb1 61fa 2461 10c1
 Package    : fedora-release-20-3.noarch (@fedora-updates/$releasever)
 From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-fedora-20-x86_64
Create subvolume '//test'
FATA[0033] Error response from daemon: Cannot destroy container c9badf5fc87bb9bfb50a3ee6e5e7c840476bd704e62404c9136aab4d27239d1e: Driver btrfs failed to remove root filesystem c9badf5fc87bb9bfb50a3ee6e5e7c840476bd704e62404c9136aab4d27239d1e: Failed to destroy btrfs snapshot: directory not empty 
```

Info:

```
# docker info
docContainers: 22
Images: 47
Storage Driver: btrfs
Execution Driver: native-0.2
Kernel Version: 3.13.2-gentoo
Operating System: Gentoo/Linux
CPUs: 8
Total Memory: 15.64 GiB
Name: whistler
ID: RL3I:O6RS:UJRN:UU74:WAGE:4X5B:T2ZU:ZRSU:BN6Q:WN7L:QTPM:VCLN
Username: phemmer
Registry: [https://index.docker.io/v1/]
WARNING: No swap limit support

# docker version
Client API version: 1.16
Go version (client): go1.3.3
OS/Arch (client): linux/amd64
Server version: 1.4.1
Server API version: 1.16
Go version (server): go1.3.3
Git commit (server): 5bc2ff8
```

### PR DESCRIPTION
**Pull request by Mashimiao**
Signed-off-by: Ma Shimiao mashimiao.fnst@cn.fujitsu.com

Fixes #9939.

### CODE REVIEW

File: integration-cli/docker_cli_rm_test.go [hunk 1]
```diff
@@ -2,11 +2,31 @@ package main
...
 
 	"github.com/go-check/check"
 )
 
+func (s *DockerSuite) TestRmContainerWithBtrfsSubvolume(c *check.C) {
+	var buf syscall.Statfs_t
+	if err := syscall.Statfs(filepath.Dir("/"), &buf); err != nil {
+		c.Fatal(err)
+	}
+	if uint32(buf.Type) != uint32(0x9123683E) {
```
Comment (cpuguy83) [inline integration-cli/docker_cli_rm_test.go]: This can go into requirements.

Comment (cpuguy83) [inline integration-cli/docker_cli_rm_test.go]: Also may be helpful to pull in `FsMagicBtrfs` from the graphdriver package.

**Comment by cpuguy83** [comment]
I'll have to setup a btrfs system to test.

**Comment by jfrazelle** [comment]
omg yay!

File: integration-cli/docker_cli_rm_test.go [hunk 2]
```diff
@@ -7,6 +7,19 @@ import (
 	"github.com/go-check/check"
 )
 
+func (s *DockerSuite) TestRmContainerWithBtrfsSubvolume(c *check.C) {
+	testRequires(c, BtrfsSystem)
+
+	out, _, err := dockerCmdWithError("run", "--name", "testwithbtrfs", "fedora:latest", "/bin/sh", "-c", "apt-get -y -q install btrfs-progs && btrfs subvolume create /test")
```
Comment (jfrazelle) [inline integration-cli/docker_cli_rm_test.go]: this test should not call fedora, please use an image we already have like busybox

File: integration-cli/requirements.go [hunk 3]
```diff
@@ -24,6 +26,16 @@ type testRequirement struct {
 var (
 	daemonExecDriver string
 
+	BtrfsSystem = testRequirement{
```
Comment (jfrazelle) [inline integration-cli/requirements.go]: this is not working because this is running on our overlay test servers

**Comment by jfrazelle** [comment]
thanks for your fix I have carried it in https://github.com/docker/docker/pull/15801

**Pull request closed without merging**